crates/rpc/src/v03/method/estimate_message_fee.rs

```diff
@@ -0,0 +1,393 @@
+        let input_json = serde_json::json!({
+            "message": {
+                "contract_address": "0x57dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374",
+                "entry_point_selector": "0xc73f681176fc7b3f9693986fd7b14581e8d540519e27400e88b8713932be01",
+                "calldata": ["0x1", "0x2"],
+            },
+            "sender_address": "0x0000000000000000000000000000000000000000",
+            "block_id": {"block_number": 1},
+        });
+        let input = EstimateMessageFeeInput::deserialize(&input_json).unwrap();
+
+        assert_eq!(
+            input,
+            EstimateMessageFeeInput {
+                message: FunctionCall {
+                    contract_address: ContractAddress::new_or_panic(felt!(
+                        "0x57dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374"
+                    )),
+                    entry_point_selector: EntryPoint(felt!(
+                        "0xc73f681176fc7b3f9693986fd7b14581e8d540519e27400e88b8713932be01"
+                    )),
+                    calldata: vec![CallParam(felt!("0x1")), CallParam(felt!("0x2")),],
+                },
+                sender_address: EthereumAddress(H160::zero()),
+                block_id: BlockId::Number(BlockNumber::new_or_panic(1)),
+            }
+        );
+    }
+
+    #[test]
+    fn test_parse_input_positional() {
+        let input_json = serde_json::json!([
+            {
+                "contract_address": "0x57dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374",
+                "entry_point_selector": "0xc73f681176fc7b3f9693986fd7b14581e8d540519e27400e88b8713932be01",
+                "calldata": ["0x1", "0x2"],
+            },
+            "0x0000000000000000000000000000000000000000",
+            {"block_number": 1},
+        ]);
+        let input = EstimateMessageFeeInput::deserialize(&input_json).unwrap();
+
+        assert_eq!(
+            input,
+            EstimateMessageFeeInput {
+                message: FunctionCall {
+                    contract_address: ContractAddress::new_or_panic(felt!(
+                        "0x57dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374"
+                    )),
+                    entry_point_selector: EntryPoint(felt!(
+                        "0xc73f681176fc7b3f9693986fd7b14581e8d540519e27400e88b8713932be01"
+                    )),
+                    calldata: vec![CallParam(felt!("0x1")), CallParam(felt!("0x2")),],
+                },
+                sender_address: EthereumAddress(H160::zero()),
+                block_id: BlockId::Number(BlockNumber::new_or_panic(1)),
+            }
+        );
+    }
+
+    #[tokio::test]
+    async fn test_estimate_message_fee() {
+        let dir = tempdir().expect("tempdir");
+        let mut db_path = dir.path().to_path_buf();
+        db_path.push("db.sqlite");
+
+        let storage = Storage::migrate(db_path, JournalMode::WAL).expect("storage");
```

Comment:
We could extract these lines into something like `Storage::temporary()` and mark that as `cfg(test)` of course?

Reply:
Done.